test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/EventSourceTest.cs

```diff
@@ -0,0 +1,71 @@
+﻿// <copyright file="EventSourceTest.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation;
+using Xunit;
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests
+{
+    public class EventSourceTest : EventListener
+    {
+        public ConcurrentQueue<EventWrittenEventArgs> Events = new ConcurrentQueue<EventWrittenEventArgs>();
+
+        public EventSourceTest()
+        {
+            this.EnableEvents(OpenTelemetryProtocolExporterEventSource.Log, EventLevel.Verbose);
+        }
+
+        [Fact]
+        public void EventSource_CheckingEventId()
```

Comment:
can you help me understand what is this unit test testing? Are we checking if 2 events use same eventid? How do we catch it when people add more events?

Reply:
@TimothyMothra can share some snippets on similar test was automated in Application Insights.

Reply:
The calls to the EventSource make it testable, so we will reach everything. This asserts:
This will check if we already processed two events with the same Id
```
Assert.DoesNotContain(eventLog.EventId, eventIds);
```

This will check if we have an attribute with value different then the writeEvent
```
var method = eventSourceType.GetMethods().FirstOrDefault(q => q.Name == eventLog.EventName);

// check mismatch eventid
Assert.NotNull(method);
```

The problem: do this automatically. Right now, i had to use manually all events. so, if someone create a new method and forget to add here, it may not catch the issue.

Reply:
> The problem: do this automatically. Right now, i had to use manually all events. so, if someone create a new method and forget to add here, it may not catch the issue.

This means this PR won't help us catch the issue. If someone adds a new method, and remembers to add it here, they can also remember to check the ids dont conflict! I am sure we solved it in ApplicationInsights. Lets see if @TimothyMothra can help here.

Reply:
he pointed out last time: https://github.com/open-telemetry/opentelemetry-dotnet/issues/864 i will check again if i can integrate this.

Reply:
just updated the branch to use the same strategy that AI uses